app/models/forms/LiabilityDetailsForm.scala

```diff
@@ -45,20 +45,22 @@ object LiabilityDetailsForm {
 
   private def validEmailFormat(email: String): Boolean = email.trim.isEmpty || emailRegex.findFirstMatchIn(email.trim).nonEmpty
 
-  private def form2Liability(existingLiability: LiabilityOrder): (Option[Instant], String, Option[String], Option[String], Option[String], Option[String], String, String, Option[String]) => LiabilityOrder = {
+  private def form2Liability(existingCase: Case): (Option[Instant], String, Option[String], Option[String], Option[String], Option[String], String, String, Option[String]) => Case = {
     case (entryDate, traderName, goodName, entryNumber, traderCommodityCode, officerCommodityCode, contactName, contactEmail, contactPhone) =>
-      existingLiability.copy(
+      existingCase.copy(application = existingCase.application.asLiabilityOrder.copy(
```

Comment:
FWIW we could try using https://github.com/softwaremill/quicklens to get rid of nested `.copy` stuff

Reply:
for me copy is more readable, I don't think we need to add new lib specifically for this stuff

maybe in future

thanks